Remove remaining ICodeLoader Interfaces (#10772)
microsoft/FluidFramework: BREAKING.md

- [Remove `noopTimeFrequency` and `noopCountFrequency` from ILoaderOptions](#remove-nooptimefrequency-and-noopcountfrequency-from-iloaderoptions)
- [proxyLoaderFactories members removed from ILoaderProps and ILoaderServices](#proxyloaderfactories-members-to-be-removed-from-iloaderprops-and-iloaderservices)
- [IContainer.connectionState yields finer-grained ConnectionState values](#icontainerconnectionstate-yields-finer-grained-connectionstate-values)

> - [Remove ICodeLoader from container-definitions](#Remove-ICodeLoader-from-container-definitions)

Reviewer: nits: [Remove ICodeLoader from `@fluidframework/container-definitions`]

> - [Remove ICodeLoader from @fluidframework/container-definitions](#Remove-ICodeLoader-from-container-definitions)

Reviewer: this is part of 1.0.0 breaking changes. this should this be included under 2.0.0 breaking change section :) 

link needs to be fixed here: 
- [Remove ICodeLoader from `@fluidframework/container-definitions`](#Remove-ICodeLoader-from-fluidframeworkcontainer-definitions)

`ConnectionState.Connecting` is marked as deprecated, please use `ConnectionState.CatchingUp` instead.
`ConnectionState.Connecting` will be removed in the following major release.

> ### Remove ICodeLoader from container-definitions

Reviewer: here as well -- - Remove ICodeLoader from `@fluidframework/container-definitions`

> `ICodeLoader` in `@fluidframework/container-definitions` was deprecated a while ago and is now removed. Use `ICodeDetailsLoader` by importing it from `@fluidframework/container-loader`.

Reviewer 2: Nits:

> `ICodeLoader` in `@fluidframework/container-definitions` was deprecated a while ago and is now removed.

Can we replace "a while ago" with the exact version? Either that or just remove it all together

```suggestion
`ICodeLoader` in `@fluidframework/container-definitions` was deprecated a while ago and is now removed. Use `ICodeDetailsLoader` from `@fluidframework/container-loader` instead.
```

> Please refer to [replace ICodeLoader with ICodeDetailsLoader interface](#Replace-ICodeLoader-with-ICodeDetailsLoader-interface) for more details.

Reviewer 2: I think it's probably better to just copy the details from the section you referenced so people don't have to go searching for it
